### PR TITLE
Correctly indicate command line options in DisabledConfigFormatter

### PR DESCRIPTION
**Pull request by agrimm**
The aim of this branch is to ensure that `.rubocop_todo.yml` correctly indicates the command line options passed to it. However, the code changes are extremely ugly, and might break custom cops. Should some alternative approach be taken, either storing the command line arguments in `RuboCop::ConfigStore`, or just using ARGV within DisabledConfigFormatter?

An apparently redundant spec was deleted in the process.

### CODE REVIEW

File: lib/rubocop/formatter/base_formatter.rb [hunk 1]
```diff
@@ -57,8 +57,12 @@ class BaseFormatter
       #
       # @param output [IO]
       #   `$stdout` or opened file
-      def initialize(output)
+      #
+      # @param command_line_args [Array(String)]
+      #   the arguments passed at the command line
+      def initialize(output, command_line_args)
```
Comment (jonas054) [inline lib/rubocop/formatter/base_formatter.rb]: Changing a public API here. That will break custom formatters.

**Comment by bbatsov** [comment]
ping :-)

**Comment by agrimm** [comment]
What's the most logical way to store what command line options are involved?

**Comment by jonas054** [comment]
> or just using ARGV within DisabledConfigFormatter

I think that's the way to go. You could define

```ruby
def initialize(output, command_line_args = ARGV)
```

only in `DisabledConfigFormatter`. Then you stay compatible in the base class (important for custom formatters), but can supply your command line arguments of choice in `disabled_config_formatter_spec.rb`.

**Comment by bbatsov** [comment]
Closing due to lack of activity.

**Pull request closed without merging**